src/realm/db.cpp

```diff
@@ -2856,3 +2865,22 @@ Replication::~Replication()
         m_db->set_replication(nullptr);
     }
 }
+
+void Transaction::async_request_sync_to_storage(std::function<void()> when_synchronized)
+{
+    m_is_synchronizing = true;
+    // get a callback on the helper thread, in which to sync to disk
+    get_db()->async_sync_to_disk([&, when_synchronized]() {
+        // sync to disk:
+        DB::ReadLockInfo read_lock;
+        db->grab_read_lock(read_lock, VersionID());
+        GroupWriter out(*this);
+        out.commit(read_lock.m_top_ref);
+        // we must release the write mutex before the callback, because the callback
+        // is allowed to re-request it.
+        db->release_read_lock(read_lock);
+        db->do_end_write();
+        when_synchronized();
```

Comment:
should we guard against exceptions when calling the provided callback. Just a thought.